subt/local_planner.py

```diff
@@ -69,4 +76,97 @@ def is_good(direction):
 
         return max((is_good(math.radians(direction)), math.radians(direction)) for direction in range(-180, 180, 3))
 
+
+class LocalPlannerOpt:
+    def __init__(self, scan_right=math.radians(-135), scan_left=math.radians(135), direction_adherence=math.radians(90), max_obstacle_distance=1.5, obstacle_influence=1.2):
+        self.last_scan = None
+        self.scan_right = scan_right
+        self.scan_left = scan_left
+        self.direction_adherence = direction_adherence
+        self.max_obstacle_distance = max_obstacle_distance
+        self.obstacle_influence = obstacle_influence
+
+    def update(self, scan):
+        self.last_scan = scan
+
+    def recommend(self, desired_dir):
+        if self.last_scan is None:
+            return 1.0, desired_dir
+
+        valid = [True] * 361  # indexed -180 .. 180
+
+        obstacles = []
+        for (i, measurement) in enumerate(self.last_scan):
+            if measurement == 0:
+                continue
+            if measurement * 1e-3 > self.max_obstacle_distance:
+                continue
+            measurement_angle = self.scan_right + (self.scan_left - self.scan_right) * i / float(len(self.last_scan) - 1)
+            measurement_vector = math.cos(measurement_angle), math.sin(measurement_angle)
+
+            # Converting from tenths of milimeters to meters.
+            obstacle_xy = [mv * measurement * 1e-3 for mv in measurement_vector]
```

Comment:
I believe the comment is not correct - the measurement is in millimeters, right?

My personal preference is to not use this "embedded for" syntax for 2 valued list, but that is a minor nitpick.

Reply:
OK, fixed comment (it was copied from old code, the same with 2 valued list - I agree)